src/main/java/raisetech/crudsample/controller/Controller.java

```diff
@@ -0,0 +1,85 @@
+package raisetech.crudsample.controller;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+import raisetech.crudsample.entity.Message;
+import raisetech.crudsample.from.MsgForm;
+import raisetech.crudsample.from.UpdateForm;
+import raisetech.crudsample.service.MsgService;
+
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/msg")
+public class Controller {
+    private final MsgService msgService;
+
+    public Controller(MsgService msgService) {
+        this.msgService = msgService;
+    }
+
+    @GetMapping
+    public List<Message> findAll() {
+        return msgService.findAll();
+    }
+
+    @GetMapping("/{id}")//idが存在しない場合は404を返す
+    public Message findById(@PathVariable int id) {
+        return msgService.findById(id);
+    }
+
+    @PostMapping //30文字を超える場合はエラーメッセージを返す
+    public ResponseEntity<Map<String, String>> createMsg(@RequestBody @Valid MsgForm msgForm, UriComponentsBuilder uriComponentsBuilder) {
+        String newId = String.valueOf(msgService.createMsg(msgForm));
+        URI url = uriComponentsBuilder.fromUriString("http://localhost:8080")
+                .path("/msg/{id}")
```

Comment:
"http://localhost:8080"のようにハードコードしないで済むように実装を修正しましょう。
```suggestion
        URI url = uriComponentsBuilder
                .path("/msg/{id}")
```
こうするとEC2にデプロイした際にも自動でhost部分が補完されるようになります。

Reply:
こちらも改善対応として別PRをあげてくれればOKです。